flink-connectors/flink-connector-hbase/src/main/java/org/apache/flink/connector/hbase/source/split/HBaseSourceSplitState.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.connector.hbase.source.split;
+
+import org.apache.flink.connector.hbase.HBaseEvent;
+import org.apache.flink.connector.hbase.source.reader.HBaseSourceEvent;
+
+/**
+ * Contains the state of an {@link HBaseSourceSplit}. It tracks the timestamp of the last emitted
+ * event to ensure no duplicates appear on recovery.
+ */
+public class HBaseSourceSplitState {
```

Comment:
Please add `@Internal` to all classes that are public (because Java doesn't know C#'s `internal`) and are not supposed to be used by the end-users.

Reply:
✅ in 702cc78e47b502ac33b67ba0328892ca8c07e293